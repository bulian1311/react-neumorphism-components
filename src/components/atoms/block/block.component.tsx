import React from 'react';
import { StyledBlock } from './block.styled';

export const Block: React.FC = ({ children }) => {
  return <StyledBlock>{children}</StyledBlock>;
};
