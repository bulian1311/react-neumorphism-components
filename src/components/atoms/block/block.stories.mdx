import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks';
import * as stories from './block.stories.tsx';
import { Block } from './block.component.tsx';

<Meta title="Atoms/Block" />

# Block

The block element its the simplest of the components, his only function its to add margin between the elements

<ArgsTable of={Block} />

## Example

<Canvas>
  <Story story={stories.Block} name="Block" />
</Canvas>
