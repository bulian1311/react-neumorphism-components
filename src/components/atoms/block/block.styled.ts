import styled from 'styled-components';

export const StyledBlock = styled.div`
  &:not(:last-child) {
    margin-bottom: 1.5rem;
  }
`;
