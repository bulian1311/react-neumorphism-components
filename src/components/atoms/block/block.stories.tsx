import React from 'react';
import { Story, Meta } from '@storybook/react';

import { Block as BlockComponent } from './block.component';

export default {
  title: 'Atoms/Block',
  component: BlockComponent,
  argTypes: {
    withoutBlock: {
      type: { name: 'boolean', required: false },
      name: 'Without Block',
      control: { type: 'boolean' },
    },
  },
} as Meta;

const Template: Story = ({ withoutBlock }) => (
  <>
    {withoutBlock ? (
      <>
        <div>This text is not within a block.</div>
        <div>This text is not within a block.</div>
        <div>This text is not within a block.</div>
      </>
    ) : (
      <>
        <BlockComponent>This text is within a block.</BlockComponent>
        <BlockComponent>This text is within a block.</BlockComponent>
        <BlockComponent>This text is within a block.</BlockComponent>
      </>
    )}
  </>
);

export const Block = Template.bind({});
Block.args = {
  withoutBlock: false,
};
