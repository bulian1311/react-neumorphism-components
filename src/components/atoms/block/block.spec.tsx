import React from 'react';
import { Block } from './block.component';
import { render, RenderResult } from '@testing-library/react';

let documentBody: RenderResult;

describe('Block', () => {
  beforeEach(() => {
    documentBody = render(<Block>Block</Block>);
  });

  it('Should Exist', () => {
    expect(Block).toBeDefined();
  });

  it('To be in the document.', () => {
    const { getByText } = documentBody;
    expect(getByText('Block')).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { container } = documentBody;
    expect(container.firstChild).toMatchSnapshot();
  });
});
