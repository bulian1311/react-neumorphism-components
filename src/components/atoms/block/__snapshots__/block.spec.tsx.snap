// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Block matches snapshot 1`] = `
<div
  class="sc-bdvvtL fXrSIV"
>
  Block
</div>
`;
